ebms-async/src/test/kotlin/no/nav/emottak/ebms/async/processing/SignalMessageServiceTest.kt

```diff
@@ -0,0 +1,94 @@
+package no.nav.emottak.ebms.async.processing
+
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.mockk
+import kotlinx.coroutines.runBlocking
+import no.nav.emottak.ebms.async.util.EventRegistrationServiceFake
+import no.nav.emottak.ebms.validation.CPAValidationService
+import no.nav.emottak.message.model.Acknowledgment
+import no.nav.emottak.message.model.EbMSDocument
+import no.nav.emottak.message.model.MessageError
+import no.nav.emottak.message.model.Payload
+import no.nav.emottak.message.xml.createDocument
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import kotlin.uuid.Uuid
+
+class SignalMessageServiceTest {
+
+    val cpaValidationService = mockk<CPAValidationService>()
+    val eventRegistrationService = EventRegistrationServiceFake()
+    val signalMessageService = SignalMessageService(cpaValidationService, eventRegistrationService)
+
+    @Test
+    fun `Acknowledgment message processed successfully`() {
+        val document = runBlocking {
+            this::class.java.classLoader
+                .getResourceAsStream("signaltest/acknowledgment.xml")!!.readAllBytes().createDocument()
+        }
+
+        val requestId = Uuid.random().toString()
+        val acknowledgment = EbMSDocument(
+            requestId = requestId,
+            dokument = document,
+            attachments = emptyList()
+        ).transform() as Acknowledgment
+
+        coEvery { cpaValidationService.validateIncomingMessage(acknowledgment) } returns mockk(relaxed = true)
+
+        runBlocking {
+            signalMessageService.processSignal(requestId, acknowledgment)
+        }
+
+        coVerify(exactly = 1) { cpaValidationService.validateIncomingMessage(acknowledgment) }
+        coVerify(exactly = 1) { signalMessageService.processAcknowledgment(acknowledgment) }
+    }
+
+    @Test
+    fun `MessageError processed successfully`() {
+        val document = runBlocking {
+            this::class.java.classLoader
+                .getResourceAsStream("signaltest/messageerror.xml")!!.readAllBytes().createDocument()
```

Comment:
Legendarisk pirking (valgfritt, trenger ikke å gjøre noe): Endre navn til `message-error.xml` for å gjøre den enklere å lese?